client/app/services/query.js

```diff
@@ -24,44 +34,75 @@ class Parameter {
     this.title = parameter.title;
     this.name = parameter.name;
     this.type = parameter.type;
-    this.value = parameter.value;
+    this.useCurrentDateTime = parameter.useCurrentDateTime;
     this.global = parameter.global;
     this.enumOptions = parameter.enumOptions;
     this.queryId = parameter.queryId;
 
-    // method to update parameter value from date/time picker component
-    // (react does not support two-way binding with `ngModel`)
-    this.updateValue = (value) => {
-      this.ngModel = value;
-    };
+    // validate value and init internal state
+    this.setValue(parameter.value);
+
+    // explicitly bind it to `this` to allow passing it as callback to Ant's
+    // DatePicker component
+    this.setValue = this.setValue.bind(this);
   }
 
-  get ngModel() {
-    if (this.type === 'date' || this.type === 'datetime-local' || this.type === 'datetime-with-seconds') {
-      this.$$value = this.$$value || moment(this.value).toDate();
-      return this.$$value;
-    } else if (this.type === 'number') {
-      this.$$value = this.$$value || parseInt(this.value, 10);
-      return this.$$value;
-    }
+  get isEmptyValue() {
+    return isNull(this.value) || isUndefined(this.value) || (this.value === '');
+  }
 
+  get isEmpty() {
+    // `isEmptyValue` may return true, but parameter may actually be not empty -
+    // in example, date/time params may have default value
+    return isNull(this.getValue());
+  }
+
+  getValue() {
+    if (this.isEmptyValue) {
+      if (
+        includes(['date', 'datetime-local', 'datetime-with-seconds'], this.type) &&
+        this.useCurrentDateTime
+      ) {
+        return moment().format(DATETIME_FORMATS[this.type]);
+      }
+      return null; // normalize empty value
+    }
+    if (this.type === 'number') {
+      return parseInt(this.value, 10) || null; // normalize empty value
+    }
     return this.value;
   }
 
-  set ngModel(value) {
-    if (value && this.type === 'date') {
-      this.value = moment(value).format('YYYY-MM-DD');
-      this.$$value = moment(this.value).toDate();
-    } else if (value && this.type === 'datetime-local') {
-      this.value = moment(value).format('YYYY-MM-DD HH:mm');
-      this.$$value = moment(this.value).toDate();
-    } else if (value && this.type === 'datetime-with-seconds') {
-      this.value = moment(value).format('YYYY-MM-DD HH:mm:ss');
-      this.$$value = moment(this.value).toDate();
+  setValue(value) {
+    if (includes(['date', 'datetime-local', 'datetime-with-seconds'], this.type)) {
+      value = moment(value);
+      if (value.isValid()) {
+        this.value = value.format(DATETIME_FORMATS[this.type]);
+        this.$$value = value.toDate();
```

Comment:
I wonder if we even need this anymore... Does Ant's datetime picker expects a date object?

Reply:
It expects `moment` instance, but I cannot set `PropTypes` for this

Reply:
I'll try to learn how custom validators work and replace this with `moment` instance